### ISSUE TITLE
Support for Catppuccin theme

### ISSUE THREAD

**Issue by LatenPath** [ghaschel/vscode-angular-html#103]
**Is your request a change or a new theme?**

- [ ] Change
- [x] Request

**What is the theme name and link?**
VSCode: [Catppuccin](https://marketplace.visualstudio.com/items?itemName=Catppuccin.catppuccin-vsc)

**Describe the suggested improvement/change**
Currently, I'm using Catppuccin (Mocha color-set):
![image](https://github.com/user-attachments/assets/5b340dea-a9c3-4763-906f-708f7e99836c)

![image](https://github.com/user-attachments/assets/591a96ab-e122-49f9-a178-787f84504697)

I tried "addLegacyColorCustomizations" command but the result is... not good.
![image](https://github.com/user-attachments/assets/1bf899f2-5b4e-4b75-a60d-a82140381ec4)
![image](https://github.com/user-attachments/assets/41c5f700-3ff6-4b14-9866-71886b161f72)
![image](https://github.com/user-attachments/assets/a7cdce8b-7794-40c1-9253-119dcf75a841)

I hope that Catppuccin themes would be supported by this awesome extension. 

Thank you so much.

**Comment by ghaschel** [comment]
On it! I'm gonna implement this and all the variations during the weekend. Don't know how I missed that one.

What do you think of this?
![code 1](https://github.com/user-attachments/assets/095297c1-cb6f-456a-89e7-9241e78672af)
![code 2](https://github.com/user-attachments/assets/8954aaa7-0873-495f-be0f-43349bd16565)

**Comment by LatenPath** [comment]
Thanks for your help. I'm new so there are a lot of things I'm not sure about. Hmm... but, I think having some more similarities between the original theme colors and your new ones would be quite helpful. I'm still bad at this but I hope you can take a look and give me advice.

**Prefix**

Scope           | Typescript           |  Angular HTML
:-------------------------:|:-------------------------:|:-------------------------:
![image](https://github.com/user-attachments/assets/eae93846-3c18-4dd1-8b54-c27b2021b30f) | ![image](https://github.com/user-attachments/assets/cc1dd206-0fbb-4e15-81c8-c5906e8affb0) | ![image](https://github.com/user-attachments/assets/f881e6ae-41fe-499a-9e4a-0e3d029a3f60)

 <details close>
  <summary>Settings</summary>
```json
{
        "scope": [
          "html-template.ng.attributes.animationtrigger.prefix",
          "html-template.ng.attributes.input-binding.animationTrigger.prefix"
        ],
        "settings": {
          "foreground": "#89B4FA"
        }
      },
```
</details>

About `html-template.ng.expression.operator.logical`, I think the default color is more familiar to me, but maybe we chose a different color for some reason.

![image](https://github.com/user-attachments/assets/2963d5be-3a79-4127-8e18-a5a61bfbe276)

And also Control flow: `html-template.ng.control-flow.keyword` to me is like `keyword.control.conditional.ts`, *haha*.
![image](https://github.com/user-attachments/assets/193d0f40-0ec6-462f-b00c-5155e548054d)

By the way, pardon me, I have a newbie question here: My VSCode doesn't apply colors to bracket pair in Control flow, what should I do to fix it?
![image](https://github.com/user-attachments/assets/b2115105-8eb1-4c24-95df-79543bb2a758)

As for the rest, they are awesome, you did a beautiful theme, and I appreciate it. Thank you so much.

**Comment by LatenPath** [comment]
Note: fixed bracket-pair color issue.

![image](https://github.com/user-attachments/assets/1371a3d7-1f88-4612-9c64-6b9dcd99c52b)

`settings.json`:
```json
  "[html]": {
    "editor.language.colorizedBracketPairs": [
      ["(", ")"],
      ["{", "}"]
    ]
  }
```

**Comment by ghaschel** [comment]
Hi, I'm sorry for the delay on this. It has been some really crazy busy weeks over here. I'll look into it right now

**Comment by ghaschel** [comment]
The issue is: The themes don't follow a specific pattern for coloring the tokenized text. So what I did for all the themes is manually creating a "link" between the theme colors and the extension grammar.
By manually defining a set of colors for specific scopes that are assigned to a set of scopes within the extension. It's like, 7 scope colors that are assigned programmatically to a bunch of different scopes. So as for now, there's no easy way to make really granular changes to a specific theme and still supporting a lot of themes. 

There is though, a big refactor that I'll eventually release that will implement semantic tokenization and will allow me to address this better and maybe even fix this issue altogether.

**Comment by ghaschel** [comment]
That said, this is how it looks when I change the logic operators to the color you mentioned. Let me know what you think.

In regards to the control flow brackets, if you could open another issue, I'd really appreciate it. I'll look into it. Apparently the brackets are not being tokenized for some reason. I'll look into that and create a config entry just for the control flow

![code](https://github.com/user-attachments/assets/e9523f45-53b7-47e5-ae6b-3f91d4c64fbc)

**Comment by LatenPath** [comment]
I think the colors are very awesome now. 

Thank you so much for your support, I really appreciate it.

By the way, about control flow bracket color. My config is simply a setting in `settings.json`:
```json
  "[html]": {
    "editor.language.colorizedBracketPairs": [
      ["(", ")"],
      ["{", "}"]
    ]
  }
```
![image](https://github.com/user-attachments/assets/258eceba-9484-4e6c-aedf-75c02f7afaac)
![image](https://github.com/user-attachments/assets/b6e56cf7-8b9c-479f-852e-1d071f27f9e5)

It's not a good idea because `editor.language.colorizedBracketPairs` will override all `editor.semanticTokenColorCustomizations` color setting.

I think that it's very hard to work around it. So, this is just an interesting discovery of mine while learning, please forget it if it's not necessary.

Anyway, thank you so much again. Can't wait to see these beautiful colors.

**Comment by ghaschel** [comment]
Just published version 2.13.0, it should be live soon and it adds support for Mocha, Macchiato, Latte and Frappé.

About the bracket color, I'd say it'd be better to stick with the native vscode setting to address that. But I'll make a note and look further into it in the next major version

Closing this, but feel free to open another one if you see something off. Have a great weekend!
